src/components/mdpDatePicker/mdpDatePicker.js

```diff
@@ -461,6 +461,54 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
                 scope.$on("$destroy", function() {
                     inputElement.off("reset input blur", onInputElementEvents);
                 });
+
+                if(scope.dateFormat == "MM/DD/YYYY") {
+                    inputElement.on("keypress", formatDateAsTyped);
+                }
+
+
+                function formatDateAsTyped(event) {
+                    var inputDate = this.value;
```

Comment:
I'm not sure what "this" is here.  It would be better to assign `this` to a variable so it is easily referenced and the scope is easier to read.